system.c

```diff
@@ -265,7 +266,35 @@ uint8_t system_execute_line(char *line)
           if ( line[++char_counter] != 0 ) { return(STATUS_INVALID_STATEMENT); }
           else { report_ngc_parameters(); }
           break;
-
+        case 'A': // Probe: Specify sensor, axis and specified directionX,
+          char_counter++;
+          uint8_t axis = line[char_counter];
+          axis = get_axis_idx(axis);
+          //Axis
+          if ((axis == N_AXIS) || (line[++char_counter] != ','))
+            return(STATUS_INVALID_STATEMENT);
+          // Sensor
+          char_counter++;
+          if (!read_float(line, &char_counter, &value))
+            return(STATUS_BAD_NUMBER_FORMAT);      
+          uint8_t sensor = (uint8_t)value;
+          // Check if sensor is valid
+          if (sensor >= e_sensor_types)
+            return(STATUS_BAD_NUMBER_FORMAT);
+          // Direction
+          if (line[char_counter] != ',')
+            return(STATUS_INVALID_STATEMENT);
+          char_counter++;
+          if (!read_float(line, &char_counter, &value))
+            return(STATUS_BAD_NUMBER_FORMAT);      
+          uint8_t dir = (uint8_t)value;
+          if (dir > 1)
+            return(STATUS_BAD_NUMBER_FORMAT);
+          probe_move_to_sensor(sensor, axis, dir);
+          // End of system command:
+          if (line[++char_counter] != 0)
+            return(STATUS_INVALID_STATEMENT);
+          break;
```

Comment:
Could you toss some line breaks in between the logical blocks here (probably before every comment)? This glob is hard to parse visually.